master/buildbot/data/changes.py

```diff
@@ -104,8 +105,12 @@ def get(self, resultSpec, kwargs):
                 changes = []
         else:
             if resultSpec is not None:
-                resultSpec.fieldMapping = self.fieldMapping
```

Comment:
I may be missing something, but why can't we use fieldMapping for `branch is not None` case? Other potential filters may still be relevant.

Reply:
Yes, I think other fields may be relevant. However, based on the table joins, some of them might have a different result which might be confusing since it will filter all the entries for a particular branch. Also, some of them might be trickier to integrate. I'll look into this in more detail.

Reply:
Based on further looking, I think it should be possible to use the resultSpec, but that would be on top of the join between the changes and buildset tables, so some filters might have a different meaning between the case where 'branch' is specified or not. Is this ok?